internal/controller/pipelinerollout_controller.go

```diff
@@ -767,19 +770,34 @@ func (r *PipelineRolloutReconciler) calPipelineNameSuffix(ctx context.Context, p
 		}
```

Comment:
Just realizing we're calling `updatePipelineRolloutStatus()` above. Generally, we've updated the in-memory value of `Status` during the reconciliation but only at the end posted the Update to K8S - is there a good reason to do it differently here?

Reply:
I'm following the deployment reconciliation model here. We want to avoid the naming collision as much as possible, if `NameCount` is already updated we want to know that and return error here.

Reply:
Got it